homeassistant/components/nest/__init__.py

```diff
@@ -167,9 +171,9 @@ async def async_handle_event(self, event_message: EventMessage) -> None:
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Nest from a config entry with dispatch between old/new flows."""
-    config_mode = config_flow.get_config_mode(hass)
-    if DATA_SDM not in entry.data or config_mode == config_flow.ConfigMode.LEGACY:
-        return await async_setup_legacy_entry(hass, entry)
+    if DATA_SDM not in entry.data:
+        hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
```

Comment:
Does this need to be a background task? Can it be awaited?

Reply:
No, does not need to be background, it can be awaited. I was following the existing patterns of these: https://github.com/search?q=repo%3Ahome-assistant%2Fcore%20%22hass.async_create_task(hass.config_entries.async_remove%22&type=code  but I don't know the rationale, nor can think of a good reason not to just await here.

Reply:
I think there used to be a deadlock which is why its a task. Maybe the deadlock isn't fixed yet